workspaces/rbac/plugins/rbac-backend/src/service/extendable-id-provider.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { PluginIdProvider } from '@backstage-community/plugin-rbac-node';
+import { PermissionDependentPluginStore } from '../database/extra-permission-enabled-plugins-storage';
+import type { Config } from '@backstage/config';
+import { union } from 'lodash';
+
+export class ExtendablePluginIdProvider {
+  // plugin ids which came from application config and PluginIdProvider
+  private readonly configurationPluginIds: string[];
+
+  constructor(
+    private readonly pluginStore: PermissionDependentPluginStore,
+    pluginIdProvider: PluginIdProvider,
+    config: Config,
+  ) {
+    const pluginIdsConfig = config.getOptionalStringArray(
+      'permission.rbac.pluginsWithPermission',
+    );
+    this.configurationPluginIds = pluginIdsConfig
+      ? union(pluginIdsConfig, pluginIdProvider.getPluginIds())
+      : pluginIdProvider.getPluginIds();
+  }
```

Comment:
Curious, it seems like we kind of just merge all of the sources together. I wonder if we should just make database be the source of truth and have the configuration file and api be available as methods to update the it. WDYT?

Reply:
In this case, the code will be more complex because we need to store the 'source' information to determine where the plugin ID came from. Anyway, we should not allow the plugin ID value to be removed from the configuration, for example.

Reply:
But I will take a look.

Reply:
You are right, I forgot about having to keep track of where the changes originate from. Never mind, I think that it should be fine for now.